python/Ganga/new_tests/GPI/TestSubjobs.py

```diff
@@ -0,0 +1,59 @@
+from GangaUnitTest import GangaUnitTest
+
+class TestSubjobs(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(TestSubjobs, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
+        setConfigOption('TestingFramework', 'AutoCleanup', 'False')
+
+    def testLargeJobSubmission(self):
+        """
+        Create lots of subjobs and submit it
+        """
+        from Ganga.GPI import Job, GenericSplitter, Local
+        j = Job()
+        j.application.exe = "sleep"
+        j.splitter = GenericSplitter()
+        j.splitter.attribute = 'application.args'
+        j.splitter.values = [ ['400'] for i in range(0, 20) ]
+        j.backend = Local()
+        j.submit()
+
+        self.assertEqual( len(j.subjobs), 20 )
+
+    def testSetParentOnLoad(self):
+        """
+        Test that the parents are set correctly on load
+        """
+        from Ganga.GPI import jobs, queues, Executable, Local
+        from Ganga.GPIDev.Base.Proxy import isType
+        import random
+        import string
+
+        def flush_full_job():
+            j = jobs(0)
+            j.comment = "Make sure I'm dirty " + ''.join( random.choice( string.ascii_uppercase) for _ in range(5))
+            j._impl._getRegistry()._flush([j])
+
+        # Make sure the main job is fully loaded
+        j = jobs(0)
+        self.assertTrue( isType(j.application, Executable) )
+        self.assertTrue( isType(j.backend, Local) )
+        self.assertEqual( j.application.exe, "sleep" )
+
+        # fire off a load of threads to flush
+        for i in range(0, 20):
+            queues.add( flush_full_job )
+
+        # Now loop over and force the load of all the subjobs
+        for sj in j.subjobs:
+            self.assertEqual( sj.splitter, None )
+            self.assertTrue( isType(sj.application, Executable) )
+            self.assertTrue( isType(sj.backend, Local) )
+            self.assertEqual( sj.application.exe, "sleep" )
+            self.assertEqual( sj.application.args, ['400'] )
+            self.assertIs( sj._impl._getRoot(), j._impl)
```

Comment:
I can certainly move the system imports but the Ganga ones I regard as part of the test so would keep them as is. 

As regards spacing, should there be space or not in parentheses? I generally prefer it but I notice that PyCharm isn't a fan. I don't really mind, just happy to be consistent :)

I had thought it was better to use `self.assert...(...)` as that's used a lot elsewhere. Is that not the case? If so, I'll switch back to just `assert ...`.